linchpin/provision/library/bkr_info.py

```diff
@@ -107,21 +101,21 @@ def _check_jobs(self, ids):
                                   'result': recipe.get('result'),
                                   'id': recipe.get('job_id')})
                 results.update({'resources': resources})
-        return results
+        return results, len(resources)
 
 
 def main():
     mod = AnsibleModule(argument_spec={
-        'ids': {'type': 'list'},
-        'skip_no_system': {'type': 'bool', 'default': False},
-        'max_attempts': {'type': 'int', 'default': 60}
+        'ids': {'type': 'dict'},
+        'provision_params': {'type': 'dict'},
     })
     beaker = BeakerTargets(mod.params)
     try:
         results = beaker.get_system_statuses()
-        mod.exit_json(hosts=results, changed=True, success=True)
     except Exception as ex:
-        mod.fail_json(msg=str(ex))
+        msg, results = ex.args
+        mod.warn(msg)
+    mod.exit_json(hosts=results, changed=True)
```

Comment:
On second thought, calling exit_json here instead of fail_json means linchpin still exits 0 even if a provision fails, which seems bad. It's probably worth reverting this bit back to the old behavior and implementing a cleanup step that ensures all outstanding beaker provision requests are cancelled if any single one fails.